### DISCUSSION TITLE
Multiple wiki-js instances synchronization thru git with pages containing non-english characters only

### DISCUSSION THREAD

**Discussion by sashabel** [requarks/wiki#4832] [Error / Bug Report]
**Describe the bug**
Pages that were created (_push_) in instance "A" of wiki-js as VisualEditor HTML files, and then were synchronized (_pull_) thru git to instance "B" such that their names containing only non-English characters (e.g. בדיקה.html ) are not seen in "pages" of instance "B".

**To Reproduce**
1. Set up two instances of wiki-js such that
      - both of them are configured to perform git synchronization (Storage->Git) that is performed on the master branch
      - both configured to support multilanguage (Locale->Multilangual Namespaces enabled, Active namespaces: Hebrew, English)
2. On instance A create a page named "בדיקה" in Hebrew (/he) locale
3. check that the file "/he/בדיקה.html" was created and synchronized into remote git master branch.
4. Connect to instance B and ensure that in its local repository the file "/he/בדיקה.html" was imported
5. Try to access to the page from instance B address:  http://instance_B_IP:port/he/בדיקה

**Expected behavior**
1. http://instance_B_IP:port/he/בדיקה is a valid page that contains data.
2. The "בדיקה" page is listed in the list of http://instance_B_IP:port/a/pages

**Actual behavior**
1. http://instance_B_IP:port/he/בדיקה page does not exist in browser and there is a message asking the user whether he/she wants to create it.
2. The "בדיקה" page is not in the list of http://instance_B_IP:port/a/pages

**Host Info (please complete the following information):**
OS: Ubuntu 18.04 x64 and Docker
Wiki.js version: 2.5.219
Database engine: PostgreSQL 11

**Comment by sashabel** [comment]
Addition: it is enough one single character in page name that is not in English in order to reproduce this behaviour

**Reply by sashabel** [reply]
As a workaround, I've used the "Import Everything" button. However, this is not a solution since I want to synchronize different instances of Wiki-Js that is based on git repository as a single source of thruth.

**Reply by sashabel** [reply]
Is there an API to this button ("Import Everything" in /a/storage)that I can call it from a script?

**Reply by NGPixel** [reply]
Everything via the admin UI is done via the GraphQL API, which you can also call from your own scripts. You'd need to inspect, via the browser network tab, what is the exact query being sent and simply recreate it in your script. You'll need to generate an API token to send in the authorization header as well.

**Comment by EricFromCanada** [comment]
I believe this issue is fixed with https://github.com/requarks/wiki/pull/5272. With quotepath left enabled, non-ASCII characters in the filename would be escaped in git's output, causing the storage engine to create new pages with the wrong URL or not at all. With the current version I'm able to push a new non-ASCII filename to a wiki's git repository and have it show up at the expected URL.
